cmake/multi_image.cmake

```diff
@@ -21,12 +21,11 @@ function(share content)
 endfunction()
 
 if(IMAGE_NAME)
-  share("set(${IMAGE_NAME}KERNEL_HEX_NAME ${KERNEL_HEX_NAME})")
+  share("set(${IMAGE_NAME}KERNEL_HEX_NAME ${KERNEL_HEX_NAME} CACHE INTERNAL \"\")")
```

Comment:
Should we have a separate `share` function/macro for this? To me the `\"\"` at the end is looking a bit confusing. There we could also document the purpose of having different ways of sharing information.

Reply:
when looking at this, I did consider if the share function itself should be a real function.

Currently all it does is to place the provided text string into a cmake file, which then gets executed, as-is.

So you could even write `share("find_package(Zephyr)")` in there, and get that executed.
Because of the place as is code in the cmake file then the `\"\"` is needed.

Initially I wanted to keep such rework outside this PR, but with #4558 we might need to do this rework immediately, in order to have proper list handling and possibility to append values to shared lists.

Reply:
Done, now implemented:
- set_shared()
- get_shared()
- generate_shared()